packages/yourloops/lib/auth/api.ts

```diff
@@ -498,6 +516,69 @@ class AuthApi extends EventTarget {
     return _.cloneDeep(this.teams);
   }
 
+  public async removeTeamMember(team: Team, userId: string): Promise<Team[]> {
+    if (this.teams === null || this.teams.length < 1) {
+      throw new Error("Empty team list!");
+    }
+
+    // eslint-disable-next-line no-magic-numbers
+    if (Math.random() < 0.2) {
+      // eslint-disable-next-line no-magic-numbers
+      await waitTimeout(500 + Math.random()*200);
+      throw new Error("A random error");
+    }
+
+    const nTeams = this.teams.length;
+    for (let i = 0; i < nTeams; i++) {
+      const thisTeam = this.teams[i];
+      if (thisTeam.id === team.id) {
+        if (Array.isArray(thisTeam.members)) {
+          const idx = thisTeam.members.findIndex((tm: TeamMember): boolean => tm.userId === userId);
+          if (idx > -1) {
+            thisTeam.members.splice(idx, 1);
+          }
+        }
+        break;
+      }
+    }
```

Comment:
I'm not find of theses methods, but since you all seems to prefer them... well I will change.
But probably not for this code !
This code will be done by the API, it is a temp code, it will be remove.
So there is no need to change it.